qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -222,6 +222,17 @@ def _phase_gradient_state() -> PhaseGradientState:
 )
 
 
+def _extract_raw_int_from_fxp(val: Fxp) -> int:
+    """extracts the bits of the Fxp as a binary number (ignoring the dot)"""
+    if isinstance(val, (int, np.integer)):
+        return int(val)
+    return int(val.bin(), 2)
+
+
+def _phase_int_to_fxp(val: int, frac_bitsize: int) -> Fxp:
+    return Fxp(val, raw=True, n_word=frac_bitsize, n_frac=frac_bitsize, signed=False)
```

Comment:
We should probably not need these two methods after the refactoring suggested below.

Reply:
refactored `on_classical_vals` and removed these